content/blog/2020-12-tokio-1-0.md

```diff
@@ -0,0 +1,150 @@
+libraries to track these changes. It could also result in confusing error
+messages when accidentally depending on multiple versions of Tokio.
+
+[announced]: https://medium.com/@carllerche/announcing-tokio-df6bb4ddb34
+[async_await]: https://tokio.rs/blog/2019-11-tokio-0-2
+
+The Tokio 1.0 release ends this churn. As part of the release, we are committing
+to providing a stable foundation for the ecosystem. We have no current plans for
+Tokio 2.0 and we commit to holding back on a Tokio 2.0 release for at least 3
+years. We also commit to maintain the Tokio 1.0 branch for a minimum of 5 years.
+Tokio will keep a rolling MSRV (minimum supported rust version) policy of 6
+months. When increasing the MSRV, the new Rust version must have been released
+at least six months ago.
+
+This stability does not mean that Tokio will stagnate. We still have a lot of
+work to do.
+
+# 2021
+
+We expect to focus on a few areas this coming year: `Stream`, `io_uring`, tracing
+integration, and the Tokio stack.
+
+## `Stream`
+
+Today, the [`tokio-stream`] crate provides asynchronous iteration utilities
+based on the Stream trait. An [RFC] to move the [`Stream` trait][Stream] from
+[`futures-core`] to the Rust standard library is pending approval.
+
+Once the standard library provides the Stream trait, Tokio’s stream utilities
+will move into the tokio crate itself.
+
+[`tokio-stream`]: https://docs.rs/tokio-stream
+[RFC]: https://github.com/rust-lang/rfcs/pull/2996
+[Stream]: https://docs.rs/futures-core/0.3/futures_core/stream/trait.Stream.html
+[`futures-core`]: https://docs.rs/futures-core/
+
+## `io_uring`
+
+[`io_uring`] is a new Linux interface providing asynchronous operations for all I/O,
+including disk, while reducing required syscalls. Currently, on Linux, Tokio
+uses the epoll interface, which notoriously does not work for disk access. To
+work around this, Tokio provides asynchronous filesystem APIs by dispatching
+synchronous operations to a thread pool. While this works, it is not optimal.
+
+By leveraging io_uring, Tokio will be able to provide genuinely asynchronous
+filesystem operations. Additionally, early benchmarks of of io_uring are very
+promising and we are hopeful these performance improvements will carry over to
+Tokio.
+
+[`io_uring`]: https://kernel.dk/io_uring.pdf
+
+## `tracing`
+
+Defining a first-class story around operating Tokio applications will be a focus
+for 2021. A big part of this story will be tooling around tracing and metrics.
+The [`tracing`] crate already provides much of the needed infrastructure. It
+implements a facade layer and data model for structured diagnostics, allowing
+both libraries and applications to emit machine-readable data that can be
+consumed in a variety of ways.
+
+Over the coming year, we will build deeper integration between tracing and the
+rest of the Tokio stack. This integration will help provide the needed
+visibility into Tokio's internals to answer questions like, "how can I tune the
```

Comment:
```suggestion
visibility into Tokio's internals to answer questions. Like, "how can I tune the
```

Reply:
I don't think "Like, ..." is a complete setence on its own?

Reply:
In this instance the full stop is intended to prevent overly long sentences. It does seem a bit odd with the quoted questions, but I do read examples formulated like that all the time in technical literature. Like what I'm doing here.

One alternative would be to not phrase it like literal questions and instead write sentences describing their intent and go from first to third person.

> Like, "how can I tune the Tokio runtime to reduce my tail latencies even further?" [..]

Into:

> Like how you can tune the Tokio runtime to reduce tail latencies, know what tasks are currently running, or which locks are the most contended.

Reply:
I'll do that